src/main/java/org/truffleruby/language/methods/SharedMethodInfo.java

```diff
@@ -91,6 +91,18 @@ public SharedMethodInfo convertMethodMissingToMethod(RubyModule declaringModule,
                 newArgs);
     }
 
+    public SharedMethodInfo createWithNewArity(Arity new_arity) {
```

Comment:
Could you rename to `withArity(Arity newArity)`? That's the pattern used e.g., in `InternalMethod`.

The argument name should be `newArity` (camelCase in Java for local variables)

Reply:
Fixed!